source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -55,6 +64,36 @@ bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connec
                  effective_policy_id);
 }
 
+LogDecision RoleBasedAccessControlEngineImpl::shouldLog(
+    const Network::Connection& connection, const Envoy::Http::RequestHeaderMap& headers,
+    const StreamInfo::StreamInfo& info, std::string* effective_policy_id) const {
+  if (!action_log_) {
+    return LogDecision::Undecided;
+  }
+
+  bool matched = false;
+
+  for (const auto& policy : policies_) {
```

Comment:
This breaks the model of the RBAC filter that each policy is evaluated sequentially and if a policy is matched early, the remaining policies are ignored.

Say if we have a config with 3 policies <DENY-1, DENY-2, LOG>, probably we should consider these kind of policies invalid and require the LOG to be before ALLOW/DENY actions.

Reply:
oh sorry I got this confused with the multiple action support. please ignore this.